multi-agent/lib/mas/resources/repository/base.py

```
@@ -85,3 +85,26 @@ def group_count(
     def delete_by_identity(self, identity: Identity) -> int:
         """Delete all resources owned by *identity*.  Returns the count of deleted documents."""
         ...
+
+    @abstractmethod
+    def find_all_builtins(
+        self,
+        category: str | None = None,
+        type: str | None = None,
```

Ruff (0.15.21)

[error] Function argument `type` is shadowing a Python builtin

(A002)